lib/Doctrine/ORM/EntityManager.php

```diff
@@ -836,8 +836,10 @@ public function getRepository($entityName)
     public function contains($entity)
     {
         return $this->unitOfWork->isScheduledForInsert($entity)
-            || $this->unitOfWork->isInIdentityMap($entity)
-            && ! $this->unitOfWork->isScheduledForDelete($entity);
+            || (
+                $this->unitOfWork->isInIdentityMap($entity)
+                && ! $this->unitOfWork->isScheduledForDelete($entity)
+            );
```

Comment:
This is questionable, and should be addressed at the coding standard level, unless for some reason what is not obvious here according to you would be obvious in other places where we combine several boolean operators.